R/input-time-series.R

```diff
@@ -1,3 +1,153 @@
+get_strats_for_plot_type <- function(
+  plot_types,
+  strats = c("total", "adult", "adult_f", "adult_m", "child")
+) {
+  grid <- expand.grid(plot_types, strats)
+  sprintf("%s_%s", grid[,1], grid[,2])
+}
+
+##' Metadata for ANC plot types used in the input time series
+##' plots. Mainly for internal use.
+##'
+##' @param anc Path to file containing ANC data or ANC data object
+##'
+##' @return List of `plot_types` and `cols_keep` (columns that are
+##'   present in the actual data out of the known list of columns)
+##'
+##' @export
+get_anc_metadata <- function(anc) {
+  ## Check if anc is object or file path
+  if(!inherits(anc, c("spec_tbl_df","tbl_df","tbl","data.frame" ))) {
+    anc <- read_anc_testing(anc)
+  }
+
+  cols_list <- c("anc_clients", "anc_known_pos", "anc_already_art",
+                 "anc_tested", "anc_tested_pos", "anc_known_neg", "births_facility")
+  cols_keep <- intersect(cols_list, colnames(anc))
+  plot_types <- c(cols_keep, "anc_total_pos", "anc_status", "anc_prevalence",
+                  "anc_art_among_known", "anc_art_coverage", "births_clients_ratio")
```

Comment:
So `cols_list` are the full list of potential indicators in the uploaded data? 

Then `cols_keep` gives us which ones are actually in the data

And `plot_types` is those and any properties derived from these?

What happens if say we don't have `births_facility` in the uploaded data. Then we cannot calculate `births_clients_ratio`? Does that matter that we're returning it in the metadata, or will this not happen?

Reply:
yep pretty much but `plot_types` doesnt necessarily have to include `cols_keep` like for example in the art metadata, from the code it seemed like there were no if statements to check `births_facility` and it was assumed so i just replicated that

Reply:
Great ok, that all sounds fine! We probably have a validation check somewhere.